plugins/finale-paper/src/main/java/com/github/maxopoly/finale/listeners/ToolProtectionListener.java

```diff
@@ -21,26 +22,29 @@ public ToolProtectionListener(FinaleSettingManager settingMan) {
     }
 
     @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = true)
-    public void onBlockBreak(BlockBreakEvent e) {
+    public void onItemDamageByPlayer(PlayerItemDamageEvent e) {
         if (!settingMan.useToolProtection(e.getPlayer().getUniqueId())) {
             return;
         }
-        ItemStack is = e.getPlayer().getInventory().getItemInMainHand();
-        if (is.getEnchantments().isEmpty()) {
+        ItemStack is = e.getItem();
+        PlayerInventory inventory = e.getPlayer().getInventory();
+        if (is.getEnchantments().isEmpty()
+            || !(inventory.getItemInMainHand() != is
+            || inventory.getItemInOffHand() != is)) {
             return;
         }
         Damageable meta = ItemUtils.getDamageable(is);
         if (meta == null) {
             return;
         }
-        int maxDura = meta.hasMaxDamage() ? meta.getMaxDamage() : is.getType().getMaxDurability();
-        int health = maxDura - meta.getDamage();
+        int health = is.getType().getMaxDurability() - meta.getDamage();
         if (health <= settingMan.getToolProtectionThreshhold(e.getPlayer().getUniqueId())) {
             for (int i = 0; i < 5; i++) {
-                e.getPlayer().sendMessage(ChatColor.RED.toString() + ChatColor.BOLD + "Your tool is almost broken");
+                e.getPlayer().sendMessage(ChatColor.RED.toString() + ChatColor.BOLD + "Your %s is almost broken".formatted(
+                    e.getItem().getType().name().replace('_', ' ').toLowerCase()
```

Comment:
The Paper server comes with the en_US locale, you can use that to get the item name